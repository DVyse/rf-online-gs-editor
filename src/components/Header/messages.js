/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.Header.header',
    defaultMessage: 'This is the Header component !',
  },
  brand: {
    id: 'app.components.Header.brand',
    defaultMessage: 'RFGSEditor',
  },
  Home: {
    id: 'app.components.Header.Home',
    defaultMessage: 'Home',
  },
  Projects: {
    id: 'app.components.Header.Projects',
    defaultMessage: 'Projects',
  },
  createProject: {
    id: 'app.components.Header.createProject',
    defaultMessage: 'Create Project',
  },
  signIn: {
    id: 'app.components.Header.signIn',
    defaultMessage: 'Sign-in',
  },
  signUp: {
    id: 'app.components.Header.signUp',
    defaultMessage: 'Sign-up',
  },
  Options: {
    id: 'app.components.Header.Options',
    defaultMessage: 'Options',
  },
  Language: {
    id: 'app.components.Header.Language',
    defaultMessage: 'Language',
  },
  Customs: {
    id: 'app.components.Header.Customs',
    defaultMessage: 'Customs',
  },
  LocalSettings: {
    id: 'app.components.Header.LocalSettings',
    defaultMessage: 'Local Settings',
  },
});
