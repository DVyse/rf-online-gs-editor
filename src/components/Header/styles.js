import { Menu } from 'semantic-ui-react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export default styled(Menu)`
  -webkit-app-region: drag;

  .item {
    -webkit-app-region: no-drag;
  }

  &.ui.menu {
    border-radius: 0;
  }
`;

export const BrandLink = styled(NavLink)``;

export const BrandContent = styled.span`
  position: relative;
`;

export const BrandVersion = styled.span`
  font-weight: bold;
  font-size: 12px;
  text-transform: lowercase;
  color: gray;
  display: block;
`;

export const UserLink = styled(NavLink)``;
export const UserUserName = styled.span``;
