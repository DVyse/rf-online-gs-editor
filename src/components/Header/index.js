/**
 *
 * Header
 *
 */

import { FormattedMessage } from 'react-intl';
import { Icon, Popup, Button, Dropdown, Flag } from 'semantic-ui-react';
import { isFunction } from 'lodash';
import { Map } from 'immutable';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import React from 'react';

import application from '~/package';
import i18n from '../../i18n';
import messages from './messages';

import Menu, {
  BrandLink,
  BrandContent,
  BrandVersion,
  UserUserName,
  UserLink,
} from './styles';

const localeFlags = { en: 'us' };

/* eslint-disable react/prefer-stateless-function */
class Header extends React.PureComponent {
  render() {
    const {
      isLoggedIn,
      currentUser,
      currentProject,
      onClickLogout,
      projectsImportsProcessingData,
    } = this.props;

    const projectsImportsIsProcessing =
      projectsImportsProcessingData &&
      projectsImportsProcessingData.isProcessing;

    return (
      <Menu inverted color="violet">
        <Popup
          trigger={
            <Menu.Item as={BrandLink} to="/" exact>
              <BrandContent>
                <FormattedMessage {...messages.brand} />
                <BrandVersion>v{application.version}</BrandVersion>
              </BrandContent>
            </Menu.Item>
          }
          content={process.env.NODE_ENV}
          size="mini"
          inverted
        />

        <Menu.Item as={NavLink} to="/projects">
          <FormattedMessage {...messages.Projects} />
        </Menu.Item>
        {currentProject && (
          <Menu.Item as={NavLink} to={`/project/${currentProject.get('id')}`}>
            <Icon name="info" />
            {currentProject.get('title')}
          </Menu.Item>
        )}
        {projectsImportsIsProcessing && (
          <Menu.Item>
            {projectsImportsProcessingData.percent.toFixed(0)}%
          </Menu.Item>
        )}
        <Menu.Menu position="right">
          <FormattedMessage {...messages.Options}>
            {message => (
              <Dropdown text={message} pointing item position="left">
                <Dropdown.Menu>
                  <Dropdown.Header>
                    <FormattedMessage {...messages.Customs} />
                  </Dropdown.Header>
                  <React.Fragment>
                    <Dropdown.Item as={NavLink} to="/">
                      <Icon name="cog" />
                      <FormattedMessage {...messages.LocalSettings} />
                    </Dropdown.Item>
                  </React.Fragment>
                  <Dropdown.Divider />
                  <Dropdown.Header>
                    <FormattedMessage {...messages.Language} />
                  </Dropdown.Header>
                  {i18n.appLocales.map(locale => (
                    <Dropdown.Item key={locale}>
                      <Flag name={localeFlags[locale] || locale} />
                      {locale}
                    </Dropdown.Item>
                  ))}
                </Dropdown.Menu>
              </Dropdown>
            )}
          </FormattedMessage>

          {isLoggedIn && (
            <React.Fragment>
              <Popup
                size="mini"
                inverted
                trigger={
                  <Menu.Item as={UserLink} to="/">
                    <UserUserName>
                      <Icon name="user" />
                      {currentUser.get('login')}
                    </UserUserName>
                  </Menu.Item>
                }
                content={currentUser.getIn(['role', 'title'])}
              />
              <Popup
                size="mini"
                inverted
                trigger={
                  <Menu.Item as={NavLink} to="/project/create">
                    <Icon name="plus" fitted />
                  </Menu.Item>
                }
                content="Create new project"
              />
              {isFunction(onClickLogout) && (
                <Menu.Item>
                  <Button color="red" onClick={onClickLogout}>
                    <Icon name="sign out" fitted />
                  </Button>
                </Menu.Item>
              )}
            </React.Fragment>
          )}
          {!isLoggedIn && (
            <React.Fragment>
              <Menu.Item as={NavLink} to="/login">
                <Icon name="sign in" />
                <FormattedMessage {...messages.signIn} />
              </Menu.Item>
              <Menu.Item as={NavLink} to="/register">
                <Icon name="signup" />
                <FormattedMessage {...messages.signUp} />
              </Menu.Item>
            </React.Fragment>
          )}
        </Menu.Menu>
      </Menu>
    );
  }
}

Header.propTypes = {
  isLoggedIn: PropTypes.bool,
  currentUser: PropTypes.instanceOf(Map),
  currentProject: PropTypes.instanceOf(Map),
  onClickLogout: PropTypes.func,
  projectsImportsProcessingData: PropTypes.shape({
    isProcessing: PropTypes.bool.isRequired,
    countTotal: PropTypes.number.isRequired,
    countCompleted: PropTypes.number.isRequired,
    countProcesses: PropTypes.number.isRequired,
    percent: PropTypes.number.isRequired,
  }).isRequired,
};

Header.defaultProps = {
  isLoggedIn: false,
  currentUser: null,
  currentProject: null,
};

export default Header;
