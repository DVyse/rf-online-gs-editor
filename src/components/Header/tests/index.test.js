import React from 'react';
import { shallow } from 'enzyme';

import Header from '../index';

const projectsImportsProcessingData = {
  isProcessing: false,
  countTotal: 0,
  countCompleted: 0,
  countProcesses: 0,
  percent: 0,
};

describe('<Header />', () => {
  it('Should exist', () => {
    const wrapper = shallow(
      <Header projectsImportsProcessingData={projectsImportsProcessingData} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
