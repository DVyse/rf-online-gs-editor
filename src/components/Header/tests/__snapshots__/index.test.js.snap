// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should exist 1`] = `
<styles
  color="violet"
  inverted={true}
>
  <Popup
    content="test"
    inverted={true}
    keepInViewPort={true}
    on="hover"
    position="top left"
    size="mini"
    trigger={
      <MenuItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "styles__BrandLink-sc-1g7oq2h-1",
              "isStatic": true,
              "rules": Array [],
            },
            "displayName": "styles__BrandLink",
            "render": [Function],
            "styledComponentId": "styles__BrandLink-sc-1g7oq2h-1",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        exact={true}
        to="/"
      >
        <ForwardRef>
          <FormattedMessage
            defaultMessage="RFGSEditor"
            id="app.components.Header.brand"
            values={Object {}}
          />
          <ForwardRef>
            v
            1.0.3
          </ForwardRef>
        </ForwardRef>
      </MenuItem>
    }
  />
  <MenuItem
    as={[Function]}
    to="/projects"
  >
    <FormattedMessage
      defaultMessage="Projects"
      id="app.components.Header.Projects"
      values={Object {}}
    />
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <FormattedMessage
      defaultMessage="Options"
      id="app.components.Header.Options"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
    <MenuItem
      as={[Function]}
      to="/login"
    >
      <Icon
        as="i"
        name="sign in"
      />
      <FormattedMessage
        defaultMessage="Sign-in"
        id="app.components.Header.signIn"
        values={Object {}}
      />
    </MenuItem>
    <MenuItem
      as={[Function]}
      to="/register"
    >
      <Icon
        as="i"
        name="signup"
      />
      <FormattedMessage
        defaultMessage="Sign-up"
        id="app.components.Header.signUp"
        values={Object {}}
      />
    </MenuItem>
  </MenuMenu>
</styles>
`;
